core/src/main/java/org/commonjava/maven/ext/core/impl/RelocationManipulator.java

```diff
@@ -128,13 +161,66 @@ private boolean apply( final Project project, final Model model ) throws Manipul
             result |= updateDependencies( relocations, project.getAllResolvedProfileDependencies( session ).get( profile ) );
 
         }
+
+        result |= updateAllPlugins( model, project, relocations );
+
+        return result;
+    }
+
+    private boolean updateAllPlugins( final Model model, final Project project, final WildcardMap<ProjectVersionRef> relocations ) throws ManipulationException
+    {
+        Map<String, Plugin> pluginMap = getManagedPluginMap( model );
+        List<PluginReference> refs =  findPluginReferences( model, project, pluginMap );
+        boolean result = updatePlugins( relocations,  refs );
+
+        pluginMap = getPluginMap( model );
+        refs = findPluginReferences( model, project, pluginMap );
+
+        result |= updatePlugins( relocations, refs );
+
+        return result;
+    }
+
+    private boolean updatePlugins( final WildcardMap<ProjectVersionRef> relocations, final List<PluginReference> dependencies ) throws ManipulationException
```

Comment:
A few questions:

- What happens if the relocation includes a version? 
- What happens if the plugin includes a version ?
- What happens if the version is a variable?

Reply:
There is definitely some issue here. I can't find what the dependency override is doing with the versions to begin with.

The code created a key from the found GA (no V) and looks up the GA in `WildcardMap<ProjectVersionRef> relocations`.

I think there must also be an issue with wildcard for `artifactId` as well.